### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MansiM578**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "name": "blog-app",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
+    "@mui/icons-material": "^5.14.1",
+    "@mui/material": "^5.14.1",
+    "@mui/styled-engine-sc": "^5.12.0",
+    "@mui/x-date-pickers": "^6.10.2",
```
Comment (amishaaggarwal) [inline package.json]: uninstall this if not being used

File: package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
+    "bootstrap": "^5.3.0",
```
Comment (amishaaggarwal) [inline package.json]: are we using bootstrap in the app? if no then kindly uninstall this as well

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import React from "react";
+import { Routes, BrowserRouter, Route } from "react-router-dom";
+import AddForm from "./pages/AddForm";
+import MainPage from "./pages/mainPage";
+import Post from "./pages/post";
+import Login from "./pages/Login";
+import Error from "./pages/Error";
+import AuthProvider from "./routing/auth";
+import RequireAuth from "./routing/RequireAuth";
+import Dashboard from "./pages/Dashboard";
```
Comment (amishaaggarwal) [inline src/App.jsx]: use absolute paths

File: src/components/BlogForm/index.jsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+import Typography from "@mui/material/Typography";
+import * as React from "react";
+import defaultImage from "assets/images/noImage.jpg";
+import useUpdateForm from "./useUpdateForm";
+
+const Blogform = () => {
+  const {
+    formData,
+    errors,
+    handleChangeA,
```
Comment (amishaaggarwal) [inline src/components/BlogForm/index.jsx]: use meaningful function names

File: src/components/BlogForm/index.jsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    >
+      {id ? (
+        <Typography variant="h5" gutterBottom sx={{ paddingBottom: 5 }}>
+          Edit Form
+        </Typography>
+      ) : (
+        <Typography variant="h5" gutterBottom sx={{ paddingBottom: 5 }}>
+          Add Form
+        </Typography>
+      )}
```
Comment (amishaaggarwal) [inline src/components/BlogForm/index.jsx]: kindly convert the above code to oneliner

File: src/components/BlogForm/index.jsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+            value={id ? updatedFormData?.paraDate || "" : formData?.paraDate}
+            type="date"
+            onChange={id ? handleChange : handleChangeA}
+            fullWidth
+            size="small"
+            variant="outlined"
+            required
+            // inputProps={{
+            //   max: formatDateForInput(new Date()), // Optional: Set the max date to prevent future dates selection
+            //   min: formatDateForInput("1900-01-01"), // Optional: Set the min date to restrict past dates
```
Comment (amishaaggarwal) [inline src/components/BlogForm/index.jsx]: remove unused code

File: src/components/BlogForm/index.jsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+                    handleImageUpload(image);
+                  };
+                  reader.readAsDataURL(file);
+                }}
+              />
+            </>
+          )}
+        </Grid>
+        <Grid item xs={12} sm={6} />
+        <Grid item xs={12} sm={5} />
```
Comment (amishaaggarwal) [inline src/components/BlogForm/index.jsx]: remove code if not required

File: src/components/BlogForm/index.jsx [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+                color: "red",
+                borderColor: "black",
+              },
+            }}
+            onClick={handleCancel}
+          >
+            Cancel
+          </Button>
+        </Grid>
+        <Grid item xs={12} sm={5} />
```
Comment (amishaaggarwal) [inline src/components/BlogForm/index.jsx:208]: remove code if not required

File: src/components/BlogForm/useUpdateForm.js [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    const { name, value } = event.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  // function to validate the form data
+  const validateFormA = () => {
+    let isValid = true;
```
Comment (amishaaggarwal) [inline src/components/BlogForm/useUpdateForm.js:37]: optimize this function

File: src/components/BlogForm/useUpdateForm.js [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+  });
+
+  const [errors, setErrors] = useState({
+    heading: "",
+    paraName: "",
+    paraDate: "",
+    content: "",
+  });
+
+  const handleChangeA = (event) => {
```
Comment (amishaaggarwal) [inline src/components/BlogForm/useUpdateForm.js]: do we need separate handleChange functions for both add and edit?

File: src/components/BlogForm/useUpdateForm.js [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+    setUpdatedFormData(updatedData);
+  };
+
+  // function to validate the form data
+  const validateForm = () => {
+    let isValid = true;
+    const newErrors = {};
+
+    // moment(formData.paraDate).format("YYYY-MM-DD");
+    console.log(updatedFormData.paraDate);
```
Comment (amishaaggarwal) [inline src/components/BlogForm/useUpdateForm.js]: remove console.logs

File: src/components/Content/useContent.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+function useContent() {
+  const [storedData, setStoredData] = useState([]);
+
+  useEffect(() => {
+    const storedFormData = localStorage.getItem("formData");
+
```
Comment (amishaaggarwal) [inline src/components/Content/useContent.js]: useEffects should be at the file end just before return 
all variable declarations should be at the topmost level In the function or desired scope

File: src/components/MainContent/index.jsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import useMainContent from "./useMainContent";
+
+function MainContent() {
+  const [storedData, handleDelete] = useMainContent();
+  return (
+    <div>
```
Comment (amishaaggarwal) [inline src/components/MainContent/index.jsx]: remove unnecessary divs

File: src/components/MainContent/useMainContent.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+      setStoredData(JSON.parse(storedFormData));
+    }
+  }, []);
+
+  const handleDelete = (id) => {
+    const storedFormData = JSON.parse(localStorage.getItem("formData"));
+
+    const updatedItems = storedFormData.filter((obj) => obj.id !== id);
+
+    localStorage.setItem("formData", JSON.stringify(updatedItems));
```
Comment (amishaaggarwal) [inline src/components/MainContent/useMainContent.js]: use the functions from storage file

File: src/components/Navbar/index.jsx [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+import Container from "@mui/material/Container";
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import MenuItem from "@mui/material/MenuItem";
+import AdbIcon from "@mui/icons-material/Adb";
+// import "bootstrap/dist/css/bootstrap.min.css";
+import { Link } from "react-router-dom";
+import "./styles.css";
+import { useAuth } from "../../routing/auth";
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx]: use absolute imports

File: src/components/Navbar/index.jsx [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+import Typography from "@mui/material/Typography";
+import Menu from "@mui/material/Menu";
+import MenuIcon from "@mui/icons-material/Menu";
+import Container from "@mui/material/Container";
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import MenuItem from "@mui/material/MenuItem";
+import AdbIcon from "@mui/icons-material/Adb";
+// import "bootstrap/dist/css/bootstrap.min.css";
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx]: remove comments

File: src/components/Navbar/index.jsx [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+                to="/dashboard"
+                style={{ color: "white", textDecoration: "none" }}
+              >
+                My Blog APP
+              </Link>
+            ) : (
+              <Link to="/" style={{ color: "white", textDecoration: "none" }}>
+                My Blog APP
+              </Link>
+            )}
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx]: convert this link to a oneliner

File: src/components/Navbar/index.jsx [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+              open={Boolean(anchorElNav)}
+              onClose={handleCloseNavMenu}
+              sx={{
+                display: { xs: "block", md: "none" },
+              }}
+            >
+              <MenuItem onClick={handleCloseNavMenu}>
+                {auth.isLoggedIn ? (
+                  <Typography textAlign="center">
+                    <Link
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx:94]: can we use && instead of ternary operator here?

File: src/components/Navbar/index.jsx [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+                      style={{ color: "black", textDecoration: "none" }}
+                    >
+                      Create a New Post
+                    </Link>
+                  </Typography>
+                ) : (
+                  ""
+                )}
+                {auth.isLoggedIn ? (
+                  <Typography textAlign="center">
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx:104]: kindly use && here

File: src/components/Navbar/index.jsx [hunk 20]
```diff
@@ -0,0 +1,252 @@
...
+                to="/dashboard"
+                style={{ color: "white", textDecoration: "none" }}
+              >
+                My Blog APP
+              </Link>
+            ) : (
+              <Link to="/" style={{ color: "white", textDecoration: "none" }}>
+                My Blog APP
+              </Link>
+            )}
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx]: convert to a one liner

File: src/components/Navbar/index.jsx [hunk 21]
```diff
@@ -0,0 +1,252 @@
...
+                <Link
+                  to="../addForm"
+                  sx={{ mx: 2 }}
+                  style={{ color: "white", textDecoration: "none" }}
+                >
+                  Create a New Post
+                </Link>
+              </Button>
+            ) : (
+              ""
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx]: use &&

File: src/components/Navbar/index.jsx [hunk 22]
```diff
@@ -0,0 +1,252 @@
...
+            )}
+            {auth.isLoggedIn ? (
+              <Button
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx:157]: use &&

File: src/components/Navbar/index.jsx [hunk 23]
```diff
@@ -0,0 +1,252 @@
...
+                </Link>
+              </Button>
+            ) : (
+              ""
+            )}
+          </Box>
+
+          {auth.isLoggedIn ? (
+            <Box sx={{ flexGrow: 0 }}>
+              <Tooltip title="Open settings">
```
Comment (amishaaggarwal) [inline src/components/Navbar/index.jsx:174]: use &&

File: src/components/Navbar/useNavbar.js [hunk 24]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../routing/auth";
+
+function useNavbar() {
+  const auth = useAuth();
+  const navigate = useNavigate();
+
+  const [anchorElNav, setAnchorElNav] = React.useState(null);
+  const [anchorElUser, setAnchorElUser] = React.useState(null);
```
Comment (amishaaggarwal) [inline src/components/Navbar/useNavbar.js]: import useStete from react instead of React.useState

File: src/pages/AddForm/index.jsx [hunk 25]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import Paper from "@mui/material/Paper";
+import Navbar from "components/Navbar";
+import Blogform from "components/BlogForm";
+
+function AddForm() {
+  return (
+    <div>
+      <Navbar />
```
Comment (amishaaggarwal) [inline src/pages/AddForm/index.jsx]: import navbar only in the main file...not in every component

File: src/pages/AddForm/useDataForm.js
Comment (amishaaggarwal) [inline src/pages/AddForm/useDataForm.js:1]: is this file required now?

File: src/pages/Dashboard.jsx [hunk 26]
```diff
@@ -0,0 +1,27 @@
+import * as React from "react";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Box from "@mui/material/Box";
+import Content from "../components/Content";
```
Comment (amishaaggarwal) [inline src/pages/Dashboard.jsx]: absolute import

File: src/pages/Login/index.jsx [hunk 27]
```diff
@@ -0,0 +1,109 @@
+import * as React from "react";
+
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import CssBaseline from "@mui/material/CssBaseline";
+import TextField from "@mui/material/TextField";
+import { Navigate } from "react-router-dom";
+// import Grid from "@mui/material/Grid";
```
Comment (amishaaggarwal) [inline src/pages/Login/index.jsx]: remove

File: src/pages/Login/index.jsx [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Typography from "@mui/material/Typography";
+
+import Copyright from "components/Copyright";
+import Navbar from "components/Navbar";
+import useLogin from "./useLogin";
+
+// TODO remove, this demo shouldn't need to reset the theme.
+
+const defaultTheme = createTheme();
```
Comment (amishaaggarwal) [inline src/pages/Login/index.jsx:16]: is this required here?

File: src/pages/Login/useLogin.js [hunk 29]
```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../routing/auth";
```
Comment (amishaaggarwal) [inline src/pages/Login/useLogin.js]: absolute import

File: src/pages/Login/useLogin.js [hunk 30]
```diff
@@ -0,0 +1,94 @@
...
+    return isValid;
+  };
+
+  // Handle form submission
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      // login actions
+      console.log("Form is valid. Submitting data:", formData);
```
Comment (amishaaggarwal) [inline src/pages/Login/useLogin.js]: remove console

File: src/pages/Title.jsx [hunk 31]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable react/require-default-props */
```
Comment (amishaaggarwal) [inline src/pages/Title.jsx:1]: remove these now if these are added to the eslintrc.json

File: src/pages/mainPage.jsx [hunk 32]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import Navbar from "components/Navbar";
+// import MainContent from "../components/MainContent";
```
Comment (amishaaggarwal) [inline src/pages/mainPage.jsx]: remove

File: src/pages/post/index.jsx [hunk 33]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-plusplus */
+import Paper from "@mui/material/Paper";
+import Navbar from "components/Navbar";
+import React from "react";
+import Blogform from "components/BlogForm";
+
+function Post() {
+  return (
+    <>
+      <Navbar />
```
Comment (amishaaggarwal) [inline src/pages/post/index.jsx]: dont import in every file

**Review by amishaaggarwal — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (MansiM578) [reply package.json]: No

File: src/components/BlogForm/index.jsx (hunk 5 above)
Comment (MansiM578) [reply src/components/BlogForm/index.jsx]: Done

File: src/components/BlogForm/index.jsx (hunk 6 above)
Comment (MansiM578) [reply src/components/BlogForm/index.jsx]: Done

File: src/components/BlogForm/index.jsx (hunk 7 above)
Comment (MansiM578) [reply src/components/BlogForm/index.jsx]: Done

File: src/components/BlogForm/index.jsx (hunk 8 above)
Comment (MansiM578) [reply src/components/BlogForm/index.jsx:208]: Done

File: src/components/BlogForm/useUpdateForm.js (hunk 11 above)
Comment (MansiM578) [reply src/components/BlogForm/useUpdateForm.js]: Done

File: src/components/BlogForm/useUpdateForm.js (hunk 10 above)
Comment (MansiM578) [reply src/components/BlogForm/useUpdateForm.js]: Here we have use two different states that's why different functions were used. As we have both states initialized as empty we can make the common state and use them. I have tried doing this. But not Working We have to have 2 different states

File: package.json (hunk 1 above)
Comment (MansiM578) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (MansiM578) [reply package.json]: Done

File: src/pages/Login/useLogin.js (hunk 29 above)
Comment (MansiM578) [reply src/pages/Login/useLogin.js]: Done

File: src/pages/Login/useLogin.js (hunk 30 above)
Comment (MansiM578) [reply src/pages/Login/useLogin.js]: Done

File: src/pages/mainPage.jsx (hunk 32 above)
Comment (MansiM578) [reply src/pages/mainPage.jsx]: Done

File: src/components/BlogForm/useUpdateForm.js (hunk 10 above)
Comment (MansiM578) [reply src/components/BlogForm/useUpdateForm.js]: Done

File: src/components/Content/useContent.js (hunk 12 above)
Comment (MansiM578) [reply src/components/Content/useContent.js]: Done

File: src/components/MainContent/index.jsx (hunk 13 above)
Comment (MansiM578) [reply src/components/MainContent/index.jsx]: Done

File: src/components/Navbar/index.jsx (hunk 16 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx]: Done

File: src/components/Navbar/index.jsx (hunk 20 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx]: Done applied ternary to link

File: src/pages/AddForm/useDataForm.js
Comment (MansiM578) [reply src/pages/AddForm/useDataForm.js:1]: No Deleted

File: src/App.jsx (hunk 3 above)
Comment (MansiM578) [reply src/App.jsx]: Done

File: src/components/Navbar/index.jsx (hunk 15 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx]: Done

File: src/components/Navbar/index.jsx (hunk 17 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx]: Done

File: src/components/Navbar/useNavbar.js (hunk 24 above)
Comment (MansiM578) [reply src/components/Navbar/useNavbar.js]: Done

File: src/pages/Dashboard.jsx (hunk 26 above)
Comment (MansiM578) [reply src/pages/Dashboard.jsx]: Done

File: src/pages/Login/index.jsx (hunk 27 above)
Comment (MansiM578) [reply src/pages/Login/index.jsx]: Done

File: src/pages/Login/index.jsx (hunk 28 above)
Comment (MansiM578) [reply src/pages/Login/index.jsx:16]: This is used in component of page

File: src/components/BlogForm/useUpdateForm.js (hunk 9 above)
Comment (MansiM578) [reply src/components/BlogForm/useUpdateForm.js:37]: Done

File: src/components/BlogForm/index.jsx (hunk 4 above)
Comment (MansiM578) [reply src/components/BlogForm/index.jsx]: Done

File: src/components/MainContent/useMainContent.js (hunk 14 above)
Comment (MansiM578) [reply src/components/MainContent/useMainContent.js]: Done

File: src/components/Navbar/index.jsx (hunk 18 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx:94]: Done

File: src/components/Navbar/index.jsx (hunk 19 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx:104]: Done

File: src/components/Navbar/index.jsx (hunk 21 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx]: Done

File: src/components/Navbar/index.jsx (hunk 22 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx:157]: Done

File: src/components/Navbar/index.jsx (hunk 23 above)
Comment (MansiM578) [reply src/components/Navbar/index.jsx:174]: Done

**Review by amishaaggarwal — APPROVED** [review]
